### PR TITLE
docs(all): Fix and improve docs

### PR DESCRIPTION
**Pull request by cdoublev**
I know that you're working on Respond and that you may have not enough time to work on RFR docs to get it clear and concise. I hope that my suggested changes may help someone better understand how RFR works. Below are a list of things I did, as the diff view may not be very helpfull:

- [x] docs/connectRoutes: move doc about the bag option from docs/migration to docs/connectRoutes.
- [x] docs/connectRoutes: improve options order based on priority and runtime order.
- [x] docs/low-level-api: add missing doc about strict option.
- [x] docs/low-level-api: fix wrong doc about NavLink component.
- [x] docs/react-native: remove history arg as argument as it's handled internally since RFR v2.

### CODE REVIEW

File: docs/low-level-api.md [hunk 1]
```diff
@@ -11,30 +11,24 @@ import { actionToPath, pathToAction } from 'redux-first-router'
 const { routesMap } = store.getState().location
 
 const path = actionToPath(action, routesMap, querySerializer)
-const action = pathToAction(path, routesMap, querySerializer, basename)
+const action = pathToAction(path, routesMap, querySerializer, basename, strict)
 ```
 
-The `querySerializer` argument is optional and the same as the one passed to `connectRoutes`.
+The `querySerializer`, `basename`, and `strict` arguments are optionals.
```
Comment (hedgepigdaniel) [inline docs/low-level-api.md]: "optional"

File: docs/low-level-api.md [hunk 2]
```diff
@@ -11,30 +11,24 @@ import { actionToPath, pathToAction } from 'redux-first-router'
...
+
+The `querySerializer` argument is the same as the one passed to `connectRoutes`.
 It defaults to undefined.
 
-The `basename` argument to `pathToAction` is optional and works the same way as the one passed to routesMap.
-It defaults to the value passed the last time `connectRoutes` was called.
+The `basename` and `strict` arguments default to the value passed the last time `connectRoutes` was called.
+The `basename` argument works the same way as the one passed to `routesMap`.
 `actionToPath` does not apply any `basename` transformation.
+The `strict` option allows an optional trailing delimiter to match.
```
Comment (hedgepigdaniel) [inline docs/low-level-api.md]: What about "When `strict` is `true`, the presence or absence of a trailing slash is required to match the route path."?

File: docs/connectRoutes.md [hunk 3]
```diff
@@ -53,107 +55,80 @@ type RouteObject = {
...
 
-* **scrollTop** - the `scrollTop` option calls `window.scrollTo(0, 0)` on route changes so the user starts each page at the top. This is a *"poor man's"* scroll
-restoration, and should be fine while developing, especially if you're using Chrome. Though hash links won't fully function. See the next option for full-on scroll restoration support.
+* **title** - a function to specify where in your Redux state tree **Redux First Router** should expect its reducer to be attached to. This can be omitted if you attach the reducer at `state.title`. **Redux First Router** will change your page title for you when the route changes, e.g. `document.title = 'foo'`. As with `location` you can provide a function here which can be useful if you need to implement custom logic to get at the title state. Example: `state => state.get('title')`.
```
Comment (hedgepigdaniel) [inline docs/connectRoutes.md]: This is wrong - `title` is ont where the RFR reducer is attached, its how to get the page title from the state.

In this case I think the original is pretty concise and accuratet: "the name of the state key for your page title or a selector function for getting it from state"

File: docs/connectRoutes.md [hunk 4]
```diff
@@ -53,107 +55,80 @@ type RouteObject = {
...
+* **location** - a function to specify where in your Redux state tree **Redux First Router** should expect its reducer to be attached to. This can be omitted if you attach the reducer at `state.location`. You can provide a function here which can be useful if you need to implement custom logic to get at the `location` state. This is especially useful for non-standard state shapes, such as Immutable.js. For example, if `state` is an instance of `Immutable.Map` you might have `state => state.get('location')`.
```
Comment (hedgepigdaniel) [inline docs/connectRoutes.md]: I think "a function" should be changed to "A string (a key in the state) or a selector"

File: docs/low-level-api.md (hunk 1 above)
Comment (cdoublev) [reply docs/low-level-api.md]: I should have been more focus in English class, sorry (adjectives should agree with their nouns in french). :(

File: docs/low-level-api.md (hunk 2 above)
Comment (cdoublev) [reply docs/low-level-api.md]: I couldn't have said it more concisely. Note: it as was a copy/paste from the description of `path-to-regexp`.

File: docs/connectRoutes.md (hunk 3 above)
Comment (cdoublev) [reply docs/connectRoutes.md]: Sorry, I wanted to make a similar description to `location`, but I forgot the opportunity of a string. It was unclearly described for the latter.

**Pull request merged**